File: Java_Programming/Competitive/Assignment_38.java
// Program which accept one number and position from user and on that bit. Return modified number.
//
// Input: 10  3
// Output: 14

import java.util.*;

class Bitwise 
{
    public int OnBit(int iNo, int iPos)
    {
        int iMask = 0x1;

        iMask = iMask<<(iPos - 1);

        if(!((iMask | iNo) == iNo))
        {
            iNo = iMask ^ iNo;
        }

        return iNo;

    }
}

class Assignment_38
{
    public static void main(String A[])
    {
        Bitwise bobj = new Bitwise();
        Scanner sobj = new Scanner(System.in);
        int iValue = 0, iLocation = 0;
        int iRet = 0;

        System.out.println("Enter number : ");
        iValue = sobj.nextInt();

        System.out.println("Enter location of bit : ");
        iLocation = sobj.nextInt();
        
        iRet = bobj.OnBit(iValue, iLocation);

        System.out.println("Modified number : "+iRet);
        

    }
}

File: C_Programming/Competitive/Assignment_38.c
// Program which accept distance in kilometre and convert it into meter.
// (1 Kilometre = 1000 Meter)
//
// Input: 5
// Output: 5000
//
// Input: 12
// Output: 12000

#include <stdio.h>

#define METER 1000

int KMtoMeter(int iNo)
{
    int iAns = 0;

    iAns = METER * iNo;

    return iAns;
}

int main()
{
    int iValue = 0;
    int iRet = 0;

    printf("Enter distance : ");
    scanf("%d",&iValue);

    iRet = KMtoMeter(iValue);

    printf("%d Km = %d Meter\n",iValue,iRet);
    return 0;
}

// Time Complexity : O(1)